### PR TITLE
Create a hook to provide the latest block information

### LINKED ISSUE #4507: Create a hook to provide the latest block information
### Description
The latest block information is used in a few calculations including the estimation of the unlocking times.
We need a hook to fetch the latest block info, and keep it updated.

### Acceptance criteria
- The hook should fetch the latest block and return the value
- It should update the data at least every 10 sec
- The functionality must be tested using unit tests
- Use the above hook in the Unlocking Form component
- BlockDetails needs to adopt with this hook in order to get currentHeight `latestBlocks.height `

### PR DESCRIPTION
**Pull request by reyraa**
### What was the problem?

This PR resolves #4507

### How was it solved?

- [x] Create useLatestBlock hook
- [x] Remove blocks redux store, corresponding actions, reducer, and middleware
- [x] Update components to use the new hook instead of Redux store
- [x] Refactor actions that use the obsolete latestBlocks array.

### How was it tested?
- [x] Ensure the app works as it did before
- [x] Update unit tests

### CODE REVIEW

File: src/modules/block/hooks/queries/useLatestBlock.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    config,
+    options: {
+      staleTime: 1000,
+    }
+  });
+
+  return {
+    ...response,
+    data: response.data?.data[0] ?? { height: 0, timestamp: 0 },
+  };
```
Comment (soroushm) [inline src/modules/block/hooks/queries/useLatestBlock.js]: lets avoid make duplication
```suggestion
  const config = {
    params: {
      limit: 1,
    },
  };
  const response = useBlocks({config, options: {
    staleTime: 1000,
  }})

  return {
    ...response,
    data: response.data?.data[0] ?? { height: 0, timestamp: 0 },
  };
```

**Review by soroushm — CHANGES_REQUESTED** [review]

File: src/modules/dpos/validator/components/EditVote/index.test.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ describe('EditVote', () => {
     jest.clearAllMocks();
 
     useDelegates.mockReturnValue({ data: mockDelegates });
-    useBlocks.mockReturnValue({ data: mockBlocks });
+    useLatestBlock.mockReturnValue({ data: mockBlocks });
```
Comment (ikem-legend) [inline src/modules/dpos/validator/components/EditVote/index.test.js]:
```suggestion
    useLatestBlock.mockReturnValue({
      data: mockBlocks.data[0],
    });
```

File: src/modules/transaction/components/RecentTransactions/RecentTransactions.test.js [hunk 3]
```diff
@@ -2,6 +2,10 @@ import React from 'react';
 import { useSelector } from 'react-redux';
 import { cryptography } from '@liskhq/lisk-client';
 import accounts from '@tests/constants/wallets';
+import { mockBlocks } from '@block/__fixtures__';
+
```
Comment (ikem-legend) [inline src/modules/transaction/components/RecentTransactions/RecentTransactions.test.js]: @eniolam1000752 Please remove unnecessary empty line

File: src/modules/wallet/components/overview/overview.test.js [hunk 4]
```diff
@@ -26,6 +27,7 @@ jest.mock('@account/hooks');
 jest.mock('@block/hooks/queries/useBlocks');
```
Comment (ikem-legend) [inline src/modules/wallet/components/overview/overview.test.js]: @eniolam1000752 `useBlocks` is no longer in use and so the mock should be removed

File: src/modules/wallet/components/overview/overview.test.js [hunk 5]
```diff
@@ -35,10 +37,15 @@ describe('Overview', () => {
...
 
-    useTokensBalance.mockReturnValue({ data: mockTokensBalance, isLoading: false, isSuccess: true });
+    useTokensBalance.mockReturnValue({
+      data: mockTokensBalance,
+      isLoading: false,
+      isSuccess: true,
+    });
     useAuth.mockReturnValue({ data: mockAuth });
     useDelegates.mockReturnValue({ data: mockDelegates });
     useBlocks.mockReturnValue({ data: mockBlocks });
```
Comment (ikem-legend) [inline src/modules/wallet/components/overview/overview.test.js:47]: @eniolam1000752 As mentioned earlier, this should be removed too

**Review by ikem-legend — APPROVED** [review]

**Review by soroushm — APPROVED** [review]

**Pull request merged**